libpod/image/image.go

```diff
@@ -1075,3 +1077,176 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
+		}
+
+		//Add image info at beginning of buffer.
+		treeData := buffer.String()
+		buffer.Reset()
+		buffer.WriteString(fmt.Sprintf("Image ID     : %s\n", imageInfo.ID[:12]))
+		buffer.WriteString(fmt.Sprintf("RepoTags     : %s\n", imageInfo.RepoTags))
+		buffer.WriteString(fmt.Sprintf("Size         : %v\n", units.HumanSizeWithPrecision(float64(imageInfo.Size), 4)))
+		buffer.WriteString(fmt.Sprintf("Image Layers :\n"))
+		buffer.WriteString(fmt.Sprintf("%s", treeData))
+
+	}
+	return &buffer, nil
+}
+
+// Builds a Map of layerID to LayerInfo ( adds links Parent layer and child layers for each layer.)
+func buildImageLayersDependencyMap(ctx context.Context, i *Image, whatrequires bool) (map[string]*LayerInfo, error) {
+	imgs, _ := i.imageruntime.store.Images()
+	layers, _ := i.imageruntime.store.Layers()
+
+	// Builds the layer dependency chain with ParentID and ChildID
+	layerInfoMap := make(map[string]*LayerInfo)
+	for _, layer := range layers {
+		existingLayer, ok := layerInfoMap[layer.ID]
+		if !ok {
+			layerInfoMap[layer.ID] = &LayerInfo{
+				ID:       layer.ID,
+				Size:     layer.UncompressedSize,
+				ParentID: layer.Parent,
+			}
+
+			//Add current Layer as child of Parent layer.
+			existingLayer, ok = layerInfoMap[layer.Parent]
+			if !ok {
+				layerInfoMap[layer.Parent] = &LayerInfo{
+					ID: layer.Parent,
+				}
+				layerInfoMap[layer.Parent].ChildID = append(layerInfoMap[layer.Parent].ChildID, layer.ID)
+			} else {
+				existingLayer.ChildID = append(existingLayer.ChildID, layer.ID)
+			}
+		} else {
+			existingLayer.ParentID = layer.ID
+			existingLayer.Size = layer.UncompressedSize
+		}
+	}
+
+	// Add the Repo Tags to Top layer of each image.
+	for _, img := range imgs {
+		e, ok := layerInfoMap[img.TopLayer]
+		if !ok {
+			return nil, fmt.Errorf("image top-layer not found in LayerList! : %s ", img.TopLayer)
+		}
+		e.RepoTags = append(e.RepoTags, img.Names...)
+	}
+	return layerInfoMap, nil
+}
+
+// Stores all parent layers using which image is built in imageInfo
+func buildImageDependencyMap(i *Image, imageInfo *InfoImage, layerMap map[string]*LayerInfo, layerID string) error {
+	if layerID == "" {
+		return nil
+	}
+	ll, ok := layerMap[layerID]
+	if !ok {
+		return fmt.Errorf("lookup error: layerid  %s, not found", layerID)
+	}
+	if err := buildImageDependencyMap(i, imageInfo, layerMap, ll.ParentID); err != nil {
+		return err
+	}
+
+	imageInfo.Layers = append(imageInfo.Layers, *ll)
+	imageInfo.Size = imageInfo.Size + ll.Size
+	return nil
+}
+
+// Stores all children layers which are created using given Image.
+func buildImageDependentsMap(i *Image, imageInfo *InfoImage, layerMap map[string]*LayerInfo, layerID string, prefixSpace int, buffer *bytes.Buffer) error {
+	if layerID == "" {
+		return nil
+	}
+	ll, ok := layerMap[layerID]
+	if !ok {
+		return fmt.Errorf("lookup error: layerid  %s, not found", layerID)
+	}
+	imageInfo.Layers = append(imageInfo.Layers, *ll)
+	imageInfo.Size = imageInfo.Size + ll.Size
```

Comment:
This computes the total size of an image _child_ subtree (not including any of the parent layers of the base image of the hierarchy!).  I can’t quite see how that is a useful value.

Reply:
Since, the output contains  child layers only, So  it the size cumulative of all layer size.
I feel, showing size of full stack(parent layers + child layers) may be confusing to user.

Reply:
Sure, showing the size of layers that are not in the output would be confusing _as well_ — but that does not mean that showing this computed value as a property ~of the image is not confusing, or is useful.